### ISSUE TITLE
0.27 fetch issue

### ISSUE THREAD

**Issue by geqiancrazy** [facebook/react-native#8175]
reload crash
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
                                                                   at com.facebook.react.common.network.OkHttpCallUtil.cancelTag(OkHttpCallUtil.java:30)
                                                                   at com.facebook.react.modules.network.NetworkingModule.onCatalystInstanceDestroy(NetworkingModule.java:141)
                                                                   at com.facebook.react.bridge.NativeModuleRegistry.notifyCatalystInstanceDestroy(NativeModuleRegistry.java:98)
                                                                   at com.facebook.react.bridge.CatalystInstanceImpl.destroy(CatalystInstanceImpl.java:218)
                                                                   at com.facebook.react.bridge.ReactContext.destroy(ReactContext.java:193)
                                                                   at com.facebook.react.ReactInstanceManagerImpl.destroy(ReactInstanceManagerImpl.java:510)
                                                                   at com.facebook.react.ReactActivity.onDestroy(ReactActivity.java:177)
                                                                   at android.app.Activity.performDestroy(Activity.java:7102)
                                                                   at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1170)
                                                                   at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5044)
                                                                   at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5089) 
                                                                   at android.app.ActivityThread.access$1700(ActivityThread.java:222) 
                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1854) 
                                                                   at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                   at android.os.Looper.loop(Looper.java:158) 
                                                                   at android.app.ActivityThread.main(ActivityThread.java:7229) 
                                                                   at java.lang.reflect.Method.invoke(Native Method) 
                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230) 
                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)

**Comment by charpeni** [comment]
Can you provide a code sample?

**Comment by ajwhite** [comment]
Having this same problem as well, however this is running in a production application and I can't really determine where this would come from. We upgraded to `0.27.2` and started seeing customers crashing with this message.

http://crashes.to/s/1ba15b76de5

```
Fatal Exception: java.lang.RuntimeException: Unable to destroy activity {com.robinpowered.rooms/com.robinrooms.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4828)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4846)
       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5105)
       at android.app.ActivityThread.access$1100(ActivityThread.java:197)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1662)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:6873)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
       at com.facebook.react.common.network.OkHttpCallUtil.cancelTag(OkHttpCallUtil.java:30)
       at com.facebook.react.modules.network.NetworkingModule.onCatalystInstanceDestroy(NetworkingModule.java:141)
       at com.facebook.react.bridge.NativeModuleRegistry.notifyCatalystInstanceDestroy(NativeModuleRegistry.java:98)
       at com.facebook.react.bridge.CatalystInstanceImpl.destroy(CatalystInstanceImpl.java:218)
       at com.facebook.react.bridge.ReactContext.destroy(ReactContext.java:193)
       at com.facebook.react.ReactInstanceManagerImpl.destroy(ReactInstanceManagerImpl.java:510)
       at com.facebook.react.ReactActivity.onDestroy(ReactActivity.java:177)
       at android.app.Activity.performDestroy(Activity.java:6784)
       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1155)
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4806)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4846)
       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5105)
       at android.app.ActivityThread.access$1100(ActivityThread.java:197)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1662)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:6873)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
```

```

**Comment by ajwhite** [comment]
It's specifically getting a `null` reference on `call.request().tag()` in https://github.com/facebook/react-native/blob/v0.27.2/ReactAndroid/src/main/java/com/facebook/react/common/network/OkHttpCallUtil.java#L30 after the OkHttp 3 upgrade.

I'm not sure if it's as simple as verifying `call.request().tag()` is not null before performing an equality check, but that's probably what it needs.

**Comment by ajwhite** [comment]
Also reported in https://github.com/facebook/react-native/issues/8009

**Comment by hupandaxue** [comment]
The problem, I also met

**Comment by ajwhite** [comment]
This has been reported fixed in `0.28` https://github.com/facebook/react-native/issues/7755
